syntax ="proto3";

message VgaState {
  uint32  misc_output             = 1;
  uint32  status                  = 2;
  uint32  sequence_index          = 3;
  bytes   sequence                = 4;
  uint32  gfx_index               = 5;
  bytes   gfx                     = 6;
  uint32  attribute_index         = 7;
  bytes   attribute               = 8;
  uint32  crtc_index              = 9;
  bytes   crtc                    = 10;
  uint32  pallete_read_index      = 11;
  uint32  pallete_write_index     = 12;
  bytes   pallete                 = 13;
}

message VbeState {
  uint32  version                 = 1;
  uint32  index                   = 2;
  bytes   registers               = 3;
}
